dataland-community-manager/src/main/kotlin/org/dataland/datalandcommunitymanager/services/DataRequestManager.kt

```diff
@@ -117,21 +139,148 @@ class DataRequestManager(
+     */
+    @Transactional
+    fun getDataRequestById(dataRequestId: String): StoredDataRequest {
+        throwResourceNotFoundExceptionIfDataRequestIdUnknown(dataRequestId)
+        val dataRequestEntity = dataRequestRepository.findById(dataRequestId).get()
+        return buildStoredDataRequestFromDataRequestEntity(dataRequestEntity)
+    }
+
+    /**
+     * Method to patch the status of a data request.
+     * @param dataRequestId the id of the data request to patch
+     * @param requestStatus the status to apply to the data request
+     * @return the updated data request object
+     */
+    @Transactional
+    fun patchDataRequest(dataRequestId: String, requestStatus: RequestStatus): StoredDataRequest {
+        throwResourceNotFoundExceptionIfDataRequestIdUnknown(dataRequestId)
+        var dataRequestEntity = dataRequestRepository.findById(dataRequestId).get()
+        logger.info("Patching Company ${dataRequestEntity.dataRequestId} with status $requestStatus")
+        dataRequestEntity.requestStatus = requestStatus
+        dataRequestRepository.save(dataRequestEntity)
+        dataRequestEntity = dataRequestRepository.findById(dataRequestId).get()
+        return buildStoredDataRequestFromDataRequestEntity(dataRequestEntity)
+    }
+
+    /**
+     * Method to get all data requests based on filters.
+     * @param dataType the framework to apply to the data request
+     * @param requestStatus the status to apply to the data request
+     * @param userId the user to apply to the data request
+     * @param reportingPeriod the reporting period to apply to the data request
+     * @param dataRequestCompanyIdentifierValue the company identifier value to apply to the data request
+     * @return all filtered data requests
+     */
+    fun getDataRequests(
+        dataType: DataTypeEnum?,
+        userId: String?,
+        requestStatus: RequestStatus?,
+        reportingPeriod: String?,
+        dataRequestCompanyIdentifierValue: String?,
+    ): List<StoredDataRequest>? {
+        var result = dataRequestRepository.findAll()
+        fun updateResult(otherList: List<DataRequestEntity>) {
+            result = result.intersect(otherList.toSet()).toMutableList()
+        }
+
+        if (dataType != null) {
+            updateResult(dataRequestRepository.findByDataTypeName(dataType.toString()))
+        }
+        if (userId != null) {
+            updateResult(dataRequestRepository.findByUserId(userId))
+        }
+        if (requestStatus != null) {
+            updateResult(dataRequestRepository.findByRequestStatus(requestStatus))
+        }
+        if (reportingPeriod != null) {
+            updateResult(dataRequestRepository.findByReportingPeriod(reportingPeriod))
+        }
+        if (dataRequestCompanyIdentifierValue != null) {
+            result = result.intersect(
+                dataRequestRepository.findByDataRequestCompanyIdentifierValue(dataRequestCompanyIdentifierValue),
```

Comment:
I think this is pretty bad. You should not fetch all stored data requests into memory, and then filter them.
The filtering should be done by the database.

Reply:
Much better now. Thx 👍